nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/impl/adapters/StringToDocRef.java

```diff
@@ -20,42 +20,23 @@
 
 import org.nuxeo.ecm.automation.OperationContext;
 import org.nuxeo.ecm.automation.TypeAdaptException;
-import org.nuxeo.ecm.automation.TypeAdapter;
-import org.nuxeo.ecm.automation.core.scripting.Scripting;
-import org.nuxeo.ecm.core.api.DocumentModel;
 import org.nuxeo.ecm.core.api.DocumentRef;
-import org.nuxeo.ecm.core.api.IdRef;
 import org.nuxeo.ecm.core.api.NuxeoException;
-import org.nuxeo.ecm.core.api.PathRef;
 
 /**
  * @author <a href="mailto:bs@nuxeo.com">Bogdan Stefanescu</a>
  */
-public class StringToDocRef implements TypeAdapter {
+public class StringToDocRef extends AbstractTypeAdapter {
 
     @Override
     public DocumentRef getAdaptedValue(OperationContext ctx, Object objectToAdapt) throws TypeAdaptException {
         try {
             String value = (String) objectToAdapt;
-            if (value.startsWith(".")) {
-                Object obj = Scripting.newExpression("Document.resolvePathAsRef(\"" + value + "\")").eval(ctx);
-                if (obj instanceof DocumentModel) {
-                    return ((DocumentModel) obj).getRef();
-                } else if (obj instanceof DocumentRef) {
-                    return (DocumentRef) obj;
-                }
-                throw new TypeAdaptException(String.format("Cannot adapt value '%s' to a DocumentRef instance", value));
-            }
-            return createRef(value);
+            return createRef(ctx, value);
         } catch (TypeAdaptException e) {
             throw e;
         } catch (NuxeoException e) {
             throw new TypeAdaptException(e);
         }
     }
-
-    public static DocumentRef createRef(String value) {
```

Comment:
This shouldn't be removed (for backward compat) but it will become available by inheritance if it's moved to `AbstractTypeAdapter`.